polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp

```diff
@@ -79,10 +90,71 @@ static bool isSmallerThanNegativeOne(Value val, DataFlowSolver &solver) {
   return (optConstVal.has_value() && optConstVal->slt(-1));
 }
 
+/// Remove conversion like operations in the incoming expression if any.
+static Value removeConversions(Value expr) {
+  if (matchPattern(expr, m_Op<arith::ExtUIOp>()) ||
+      matchPattern(expr, m_Op<arith::ExtSIOp>()) ||
+      matchPattern(expr, m_Op<arith::TruncIOp>()) ||
+      matchPattern(expr, m_Op<arith::IndexCastOp>()) ||
+      matchPattern(expr, m_Op<arith::IndexCastUIOp>()))
+    return expr.getDefiningOp()->getOperand(0);
+  return expr;
+}
```

Comment:
Should we be abstracting more conversions here, e.g., `arith.fptosi`? Should we match `CastOpInterface` instead?

Reply:
Now checking the value is a `CastOpInterface `